### PR TITLE
#19210 Allow to reset sorting on third click on the column header in column viewer

### LINKED ISSUE #19210: Want to enhance the sorting function
**The content is translated by Google, please forgive me if the description is wrong**

When you open the database to view all tables, or view table properties, you can sort by display or column name by clicking the table header, but once sorted, you can only sort in ascending or descending order, and the original order cannot be restored. I hope that the sorting function can be enhanced. Revert sorting in addition to ascending and descending

image

image

### PR DESCRIPTION
**Pull request by E1izabeth**
Please check Database Tasks view also and on Linux and Mac.

Could be tested on PostgreSQL 12 dvdrental 

https://github.com/dbeaver/dbeaver/assets/28875055/3e670797-aaaf-41b8-b8d6-cc2cefec6f3f

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java [hunk 1]
```diff
@@ -840,7 +841,17 @@ public void handleEvent(Event e) {
             Item column = (Item)e.widget;
             if (prevColumn == column) {
                 // Set reverse order
-                sortDirection = sortDirection == SWT.UP ? SWT.DOWN : SWT.UP;
+                if (viewer.getComparator() == null) {
+                    viewer.setComparator(comparator);
+                }
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java]: Why are you removing the comparator? Just toggle between UP/DOWN/NONE sort directions.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java]: setting to NONE doesn't help

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java [hunk 2]
```diff
@@ -840,11 +840,16 @@ public void handleEvent(Event e) {
             Item column = (Item)e.widget;
             if (prevColumn == column) {
                 // Set reverse order
-                sortDirection = sortDirection == SWT.UP ? SWT.DOWN : SWT.UP;
+                if (sortDirection == SWT.NONE) {
+                    sortDirection = SWT.UP;
+                } else if (sortDirection == SWT.UP) {
+                    sortDirection = SWT.DOWN;
+                } else {
+                    sortDirection = SWT.None;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java]:
```suggestion
                    sortDirection = SWT.NONE;
```

**Review by ShadelessFox — APPROVED** [review]

**Comment by MashaKorax** [comment]
verified (tested on win/mac/linux)

**Pull request merged**